app/com/linkedin/drelephant/exceptions/ExceptionCategorization.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package com.linkedin.drelephant.exceptions;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.log4j.Logger;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+
+/**
+ * This class parsed EFCategorizationConf.xml and create List<ExceptionCategorizationData>
+ */
+
+public class ExceptionCategorization {
+  private static final Logger logger = Logger.getLogger(ExceptionCategorization.class);
+  private Map<String, List<ExceptionCategorizationData>> applicationTypeExceptionCategorizationData;
+  boolean debugEnabled = logger.isDebugEnabled();
+
+  private enum ClassificationTag {APPLICATIONTYPE, RULENAME, RULETRIGGER, RULEPRIORITY, CATEGORY}
+
+  public ExceptionCategorization(Element element) {
```

Comment:
You can create a Util method for parsing and obtaining exception categories conf data named like `loadExceptionCategorizationConf` just like done for loading Fetcher, Heuristics, etc confs. This class in only serving this one time purpose.

Reply:
This is created with similar analogy of HeuristicConfiguration and HeuristicConfigurationData .

Reply:
Seems like `ExceptionCategorizationConfiguration` would be more appropriate name, as of now in first glance it appears to be the class for Categorizing the Exceptions.